postCore/src/main/java/com/hamdan/forzenbook/post/core/viewmodel/BasePostViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.hamdan.forzenbook.post.core.viewmodel
+
+import androidx.lifecycle.ViewModel
+import com.hamdan.forzenbook.post.core.domain.PostStringValidationUseCase
+
+abstract class BasePostViewModel(
+    val postStringValidationUseCase: PostStringValidationUseCase,
+) : ViewModel() {
+    data class PostState(
+        val postText: String = "",
+        val toggled: Boolean = false,
```

Comment:
This isn't descriptive of the real state and assumes the boolean nature of your UI implementation. It should be an enum of which post type you're making